libhoney.go

```diff
@@ -502,6 +504,8 @@ func (e *Event) Send() error {
 // return an error.  Required fields are APIHost, WriteKey, and Dataset. Values
 // specified in an Event override Config.
 func (e *Event) SendPresampled() error {
+	e.lock.RLock()
+	defer e.lock.RUnlock()
```

Comment:
if this is only to protect the `len(e.data)`, thoughts on the tradeoffs around relying on `defer` here? the body of `SendPresampled` does a whole lot more (unrelated) work after the simple `len(e.data) == 0` check.

Part of my brain wants to make the argument that the event gets serialized downstream, so we **might** want that downstream behavior protected by the `RLock`, too -- except that that serialization happens after `muster` pulls it off of a channel, so any locking we do upstream of that `muster` wouldn't protect from concurrent access there anyway, as long as we're passing pointers. =\